tools/calyx-ffi-macro/src/lib.rs

```diff
@@ -0,0 +1,404 @@
+use std::{env, path::PathBuf};
+
+use parse::{CalyxFFIMacroArgs, CalyxPortDeclaration};
+use proc_macro::TokenStream;
+use quote::{format_ident, quote};
+use syn::{parse_macro_input, spanned::Spanned};
+
+mod calyx;
+mod parse;
+mod util;
+
+#[proc_macro_attribute]
+pub fn calyx_ffi(attrs: TokenStream, item: TokenStream) -> TokenStream {
```

Comment:
Please add a doc comment here to provide a high level description of what this macro does.